sched/Kconfig

```diff
@@ -1828,12 +1828,14 @@ menu "Stack and heap information"
 
 config DEFAULT_TASK_STACKSIZE
 	int "The default stack size for tasks"
+	default 65536 if ARCH_SIM
```

Comment:
please don't add sim specific setting in common place.

Reply:
if/once this revert is merged, we can consider introducing an indirection like ARCH_USE_xxx or whatever.
i personally feel it's simpler to have if ARCH_SIM here though.

Reply:
No, it's wrong to add the arch specific code in common place. Let me highlight INVIOLABLES rule here again:
https://github.com/apache/incubator-nuttx/blob/master/INVIOLABLES.md#modular-architecture